cpp/src/interop/to_arrow_device.cu

```diff
@@ -0,0 +1,590 @@
+#pragma unroll BIT_WIDTH_RATIO - 1
+                     for (auto i = 1; i < BIT_WIDTH_RATIO; ++i) {
+                       out[out_idx + i] = in[in_idx] < 0 ? -1 : 0;
+                     }
+                   });
+
+  tmp->length     = input.size();
+  tmp->null_count = input.null_count();
+
+  auto contents = input.release();
+  if (contents.null_mask) { set_buffer(contents.null_mask, 0, tmp.get()); }
+  set_buffer(buf, 1, tmp.get());
+
+  NANOARROW_RETURN_NOT_OK(
+    ArrowArrayFinishBuilding(tmp.get(), NANOARROW_VALIDATION_LEVEL_MINIMAL, nullptr));
+  ArrowArrayMove(tmp.get(), out);
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device::operator()<numeric::decimal32>(cudf::column& column,
+                                                             cudf::type_id,
+                                                             rmm::cuda_stream_view stream,
+                                                             rmm::mr::device_memory_resource* mr,
+                                                             struct ArrowArray* out)
+{
+  using DeviceType = int32_t;
+  return unsupported_decimals_to_arrow<DeviceType>(
+    column, cudf::detail::max_precision<DeviceType>(), stream, mr, out);
+}
+
+template <>
+int dispatch_to_arrow_device::operator()<numeric::decimal64>(cudf::column& column,
+                                                             cudf::type_id id,
+                                                             rmm::cuda_stream_view stream,
+                                                             rmm::mr::device_memory_resource* mr,
+                                                             struct ArrowArray* out)
+{
+  using DeviceType = int64_t;
+  return unsupported_decimals_to_arrow<DeviceType>(
+    column, cudf::detail::max_precision<DeviceType>(), stream, mr, out);
+}
+
+template <>
+int dispatch_to_arrow_device::operator()<numeric::decimal128>(cudf::column& column,
+                                                              cudf::type_id id,
+                                                              rmm::cuda_stream_view stream,
+                                                              rmm::mr::device_memory_resource* mr,
+                                                              struct ArrowArray* out)
+{
+  nanoarrow::UniqueArray tmp;
+  NANOARROW_RETURN_NOT_OK(ArrowArrayInitFromType(tmp.get(), NANOARROW_TYPE_DECIMAL128));
+  tmp->length     = column.size();
+  tmp->null_count = column.null_count();
+
+  auto contents = column.release();
+  if (contents.null_mask) { set_buffer(contents.null_mask, 0, tmp.get()); }
+
+  set_buffer(contents.data, 1, tmp.get());
+
+  NANOARROW_RETURN_NOT_OK(
+    ArrowArrayFinishBuilding(tmp.get(), NANOARROW_VALIDATION_LEVEL_MINIMAL, nullptr));
+  ArrowArrayMove(tmp.get(), out);
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device::operator()<bool>(cudf::column& column,
```

Comment:
In this case we *do* need the specialization because we have to call the `bools_to_mask` function to fetch the data instead of just being able to use the buffer as is because cudf uses 1 byte per bool and Arrow represents bools via a bitmap. 

This *could* be consolidated if desired, it would just require the extra branching if statement to call `bools_to_mask` if we're a bool array in the base default overload. that's all